app/plugin_api.go

```diff
@@ -369,6 +369,39 @@ func (api *PluginAPI) GetUserCustomStatus(userID string) (*model.CustomStatus, *
 	return api.app.GetCustomStatus(userID)
 }
 
+func (api *PluginAPI) RestoreToPreviousCustomStatus(userID string) *model.AppError {
+	cs, appErr := api.app.GetCustomStatus(userID)
+	if appErr != nil {
+		return appErr
+	}
+
+	// If user changed the custom status explicitly, don't restore to previous status
+	if cs == nil || !cs.SetByProduct {
+		return model.NewAppError("RestoreToPreviousCustomStatus", "plugin.api.restore_to_previous_custom_status", nil, "", http.StatusBadRequest)
+	}
+
+	rcs, err := api.app.GetRecentCustomStatuses(userID)
+	if err != nil {
+		return model.NewAppError("RestoreToPreviousCustomStatus", "plugin.api.restore_to_previous_custom_status", nil, "failed to get recent statuses to reset", http.StatusInternalServerError)
+	}
```

Comment:
We are losing the error detail here, I'd suggest using `Wrap(error)` method for the `model.AppError`.

Reply:
Added the method to the AppError